File: Meetme-Backend/MeetMe.Core/Persistence/Entities/EventType.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace MeetMe.Core.Persistence.Entities
{
    public class EventType
    {

        public EventType()
        {
            Questions = new List<EventTypeQuestion>();
        }
        public Guid Id { get; set; }
        public Guid OwnerId { get; set; }
        public string Name { get; set; } = null!;
        public string? Description { get; set; } 
        public string? Location { get; set; }
        public string Slug { get; set; } = null!;
        public string EventColor { get; set; }=null!;
        public bool ActiveYN { get; set; }
        public int TimeZoneId { get; set; }
        public string DateForwardKind { get; set; } = null!;
        public int? ForwardDuration { get; set; }
        public DateTime? DateFrom { get; set; }
        public DateTime? DateTo { get; set; }
        public int Duration { get; set; }
        public int BufferTimeBefore { get; set; }
        public int BufferTimeAfter { get; set; }
        public string? CustomAvailability { get; set; }
        
        public Guid? AvailabilityId { get; set; }
        public Guid CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public Guid? UpdatedBy { get; set; }

        [JsonIgnore]
        public List<EventTypeQuestion> Questions { get; set; }
        [JsonIgnore]
        public List<EventTypeAvailabilityDetail>  EventTypeAvailabilityDetails { get; set; }

        //[JsonIgnore]
        //public Availability Availability { set; get; }

    }

    public class EventTypeScheduleInfo
    {
        public EventTypeScheduleInfo()
        {
            WeeklyTimeSchedule = new List<DailyTimeSchedule>();
        }

        public Guid EventTypeId { get; set; }
        public List<DailyTimeSchedule> WeeklyTimeSchedule { get; set; }

    }
}

File: MeetMe-Frontend/projects/meetme/src/app/models/eventtype.ts
import { IAvailability } from "./IAvailability";
import { IAvailabilityDetails } from "./IAvailabilityDetails";

export interface EventType {
    id: string,
    name: string,
    description: string,
    location: string,
    eventColor: string,
    slug: string,
    activeYN: boolean,
    ownerId: string
    availabilityId:string,
    forwardDuration?:number,
    dateForwardKind: string,
    dateFrom?: Date,
    dateTo?: Date,
    duration: number,
    bufferTimeBefore: number,
    bufferTimeAfter: number,
    timeZoneId: number
    
}

export interface ICreateEventTypeCommand {
    name: string;
    description?: string;
    location?: string;
    slug: string;
    eventColor: string;
    timeZoneName:string
    activeYN: boolean;
}

export interface IUpdateEventAvailabilityCommand {
    id: string,
    dateForwardKind: string,
    forwardDuration?: number,
    dateFrom?: Date,
    dateTo?: Date,
    duration: number,
    bufferTimeBefore: number,
    bufferTimeAfter: number,
    availabilityId?: string,
    timeZoneId: number
    availabilityDetails: Array<EventAvailabilityDetailItemDto>
}

export interface EventAvailabilityDetailItemDto {

    /// <summary>
    /// D:Date
    /// W:Weekday
    /// </summary>
    dayType: string,
    value?: string,
    stepId: number,
    from: number,
    to: number
}

export interface IUpdateEventCommand {
    id: string,
    name: string,
    description?: string,
    location?: string,
    slug: string,
    eventColor: string
}

export interface IEventTypeQuestionList {
    eventTypeId: string,
    questions: IEventTypeQuestion[]
}

export interface IUpdateEventQuestionCommand {
    eventTypeId: string,
    questions: IEventTypeQuestion[]
}
export interface IEventTypeQuestion {
    id?: string,
    name: string,
    eventTypeId?: string,
    questionType: string,
    options: string,
    otherOptionYN: boolean,
    activeYN: boolean,
    requiredYN: boolean,
    displayOrder: number
}

export interface TimeZoneData {
    id: number;
    name: string;
    countryName: string;
    countryCode: string;
    cffset: string;
    offsetMinutes: number;
  }
  